kubernetes-itests/src/test/java/io/fabric8/kubernetes/PodIT.java

```diff
@@ -289,16 +291,36 @@ void uploadFile(String uploadPath) throws IOException {
     final PodResource podResource = client.pods().withName("pod-standard");
     podResource.waitUntilReady(POD_READY_WAIT_IN_MILLIS, TimeUnit.SECONDS);
     // When
-    final boolean uploadResult = podResource.file(uploadPath).upload(tempFile);
-    // Then
-    assertThat(uploadResult).isTrue();
+    retryUpload(() -> podResource.file(uploadPath).upload(tempFile));
+
     try (InputStream checkIs = podResource.file(uploadPath).read();
         BufferedReader br = new BufferedReader(new InputStreamReader(checkIs, StandardCharsets.UTF_8))) {
       String result = br.lines().collect(Collectors.joining(System.lineSeparator()));
       assertEquals("I'm uploaded", result, () -> checkFile(podResource, null, uploadPath));
     }
   }
 
+  void retryUpload(BooleanSupplier operation) {
+    Awaitility.await().atMost(60, TimeUnit.SECONDS).until(operation::getAsBoolean);
+  }
```

Comment:
I'm not really convinced about this, might eventually hide something else.

Reply:
Certainly as we are only dealing with the boolean return so this does not qualify exactly the nature of the failed upload.  However most other problems are likely to be deterministic, in which case we'll exceed the timeout.  Only other somewhat rare non-deterministic problems would be missed.